src/reusable-blocks/class-reusable-blocks-batch-handler.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace WPML\PB\Gutenberg;
+
+class Reusable_Blocks_Batch_Handler {
+
+	/** @var Reusable_Blocks */
+	private $reusable_blocks;
+
+	/** @var Reusable_Blocks_Translation */
+	private $reusable_blocks_translation;
+
+	public function __construct(
+		Reusable_Blocks $reusable_blocks,
+		Reusable_Blocks_Translation $reusable_blocks_translation
+	) {
+		$this->reusable_blocks             = $reusable_blocks;
+		$this->reusable_blocks_translation = $reusable_blocks_translation;
+	}
+
+	public function add_blocks( \WPML_TM_Translation_Batch $batch ) {
+		$blocks = \collect( $batch->get_elements() )
+			->map( [ $this, 'find_blocks_in_batch_element' ] )
+			->flatten( 1 )
+			->unique( 'block_id' );
+
+		/**
+		 * [
+		 *  [
+		 *      'block_id'     => 1,
+		 *      'target_langs' => ['fr' => 1, 'de' => 1],
+		 *      'source_lang'  => 'en',
+		 *  ],
+		 *  [
+		 *      'block_id'     => 2,
+		 *      'target_langs' => ['de' => 1],
+		 *      'source_lang'  => 'en',
+		 *  ],
+		 * ]
```

Comment:
I will do this in the next PR https://github.com/OnTheGoSystems/wpml-page-builders-gutenberg/pull/37 if you don't have objection @andreasciamanna (it will be easier since I refactored a good part of the code).

Reply:
👍

Reply:
Done here https://github.com/OnTheGoSystems/wpml-page-builders-gutenberg/pull/37/commits/941bfdcbb3b19acd042ff264e43fa44d5c40a68d.

Reply:
🎉